lib/livebook_web/live/session_live.ex

```diff
@@ -26,6 +26,20 @@ defmodule LivebookWeb.SessionLive do
             Session.get_data(session_pid)
           end
 
+        socket =
+          if connected?(socket) do
+            payload = %{
+              clients:
+                Enum.map(data.clients_map, fn {client_pid, user_id} ->
+                  client_info(client_pid, data.users_map[user_id])
+                end)
+            }
+
+            push_event(socket, "session_init", payload)
```

Comment:
Sorry for a silly question, where is this event handled? I noticed we removed `def handle_event("session_init"` elsewhere in the PR.

Reply:
It's the flow that changed :D

Previously we would load the page, mount the JS hook and from there we would `pushEvent("session_init", ...)` to the server, so we needed `handle_event`. We effectively did an additional roundtrip for the initial data.

Now we do the `push_event` directly from the server as part of the initialization, so it's transferred together with the render payload. Now we do `handleEvent("session_init", ...)` on the client to receive this data. This works, because LV first mounts the JS hooks and only then dispatches the events :)

Reply:
beautiful, thanks!